### PR TITLE
fix(core): fix snapshot restore to correctly restore tables when snapshot is taken under heavy DDL load

### PR DESCRIPTION
**Pull request by ideoma**
Fix issue found by fuzz tests when `_meta` file may not exist at the snapshot's table folder. This can occur when a table undergoes heavy column add/remove while snapshot files are copied.

Also, the PR contains a minor change to support snapshot id in .txt file. File _snapshot with instance id where the snapshot is created from has an inconvenient format, it's strictly saying a binary file.

This makes it hard to restore from the snapshots created by the instances where id was not set. This PR adds support of _snapshot.txt file which will be used if file _snapshot is empty and it can be added as a text.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 40 / 46 (86.96%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|27|33|81.82%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|13|13|100.00%|

File: core/src/main/java/io/questdb/cairo/DatabaseSnapshotAgentImpl.java [hunk 1]
```diff
@@ -390,7 +408,12 @@ void recoverSnapshot() {
             memFile.smallFile(ff, srcPath, MemoryTag.MMAP_DEFAULT);
 
             final CharSequence currentInstanceId = configuration.getSnapshotInstanceId();
-            final CharSequence snapshotInstanceId = memFile.getStr(0);
+            CharSequence snapshotInstanceId = memFile.getStr(0);
+            if (Chars.empty(snapshotInstanceId)) {
+                srcPath.trimTo(snapshotRootLen).concat(TableUtils.SNAPSHOT_META_FILE_NAME_TXT).$();
+                snapshotInstanceId = TableUtils.readText(ff, srcPath);
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/cairo/DatabaseSnapshotAgentImpl.java:414]: Might be a good idea to trim trailing spaces and newlines from the read text.

Comment (ideoma) [reply core/src/main/java/io/questdb/cairo/DatabaseSnapshotAgentImpl.java]: Added .trim()

**Review by puzpuzpuz — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 42 / 48 (87.50%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|29|35|82.86%|
|:large_blue_circle:|io/questdb/cairo/TableUtils.java|13|13|100.00%|

**Pull request merged**